astroid/nodes/node_classes.py

```diff
@@ -4667,6 +4667,34 @@ def get_children(self):
         if self.format_spec is not None:
             yield self.format_spec
 
+    def _infer(
+        self, context: InferenceContext | None = None, **kwargs: Any
+    ) -> Generator[InferenceResult, None, InferenceErrorInfo | None]:
+        if self.format_spec is None:
+            yield from self.value.infer(context, **kwargs)
+            return
+        uninferable_already_generated = False
+        for format_spec in self.format_spec.infer(context, **kwargs):
+            if not isinstance(format_spec, Const):
+                if not uninferable_already_generated:
+                    yield util.Uninferable
+                    uninferable_already_generated = True
```

Comment:
Why no do a `return` here and drop the constant?

Reply:
There are a number of scenarios (such as loops) where `inferred` will return not one but multiple values. We don't want to drop the properly inferred values just because one (or more) of them can't be inferred.

Reply:
Yes, you are completely right.